projects/igniteui-angular/src/lib/services/overlay/position/elastic-position-strategy.ts

```diff
@@ -9,46 +8,29 @@ import { Size, HorizontalAlignment, VerticalAlignment, PositionSettings } from '
 export class ElasticPositionStrategy extends BaseFitPositionStrategy {
     protected fitHorizontal(element: HTMLElement, settings: PositionSettings, innerRect: ClientRect, outerRect: ClientRect) {
         element.classList.add('igx-overlay__content--elastic');
-        let extend = 0;
-        switch (settings.horizontalDirection) {
-            case HorizontalAlignment.Left: {
-                extend = outerRect.left - innerRect.left;
-                if (extend > innerRect.width - this.settings.minSize.width) {
-                    extend = innerRect.width - this.settings.minSize.width;
-                }
-                break;
-            }
-            case HorizontalAlignment.Right: {
-                extend = innerRect.right - outerRect.right;
-                if (extend > innerRect.width - this.settings.minSize.width) {
-                    extend = innerRect.width - this.settings.minSize.width;
-                }
-                break;
-            }
-        }
+        const minExtend = innerRect.width - this.settings.minSize.width;
+        const leftExtend = Math.min(minExtend, outerRect.left - innerRect.left);
+        const rightExtend = Math.min(minExtend, innerRect.right - outerRect.right);
+        const extend = Math.max(leftExtend, rightExtend);
```

Comment:
@wnvko So if the content is in the center (what the issue is about) and it's wider than the view port (`outerRect`) it should overflow in both directions, right? So, you'll have both left and right overflow values, but this will take just the bigger one. Leaving the content element still wider than the viewport.
(same applies for top/bottom)

Reply:
fixed